### PR TITLE
Api Token module api documentation[MAD-8829]

### PR DESCRIPTION
**Pull request by Neeraja82**
Api Token module api documentation[MAD-8829]

[MAD-8829]: https://liquidweb.atlassian.net/browse/MAD-8829?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Neeraja82** [comment]
I will recheck all the files once again and make changes, based on the review comments received for App Module

**Comment by Neeraja82** [comment]
Pushed some changes. Please review.

**Review by ppf — APPROVED** [review]
for me looks ok

File: api-token/add.md [hunk 1]
```diff
@@ -1,66 +1,51 @@
...
-------------------
+# Portal: Api Token
 
-**since** 0.0.0
+## api-token:add
+Creates a new Api Token.
 
-api-token:add
-=============
+This task is queued, meaning it will be completed out-of-band from the current request. The response payload will describe the requested task, and will also include a Location header that can be polled to determine the status of the task. @see task:show.
```
Comment (adrian-enspired) [inline api-token/add.md:6]: this endpoint is an exception to the rule - it is **not** queued; the caller must wait for the response.
remove this text.

File: api-token/add.md [hunk 2]
```diff
@@ -1,66 +1,51 @@
...
 
-Creates a new api token.
+#### Access
+api-token edit
 
-The new api token is returned in the response payload. This is the only time it will be accessible to the user; it will never be displayed again.
```
Comment (adrian-enspired) [inline api-token/add.md:11]: this needs to be restored under the endpoint description above.

File: api-token/add.md [hunk 3]
```diff
@@ -1,66 +1,51 @@
...
+## api-token:add
+Creates a new Api Token.
 
-api-token:add
-=============
+This task is queued, meaning it will be completed out-of-band from the current request. The response payload will describe the requested task, and will also include a Location header that can be polled to determine the status of the task. @see task:show.
 
-Creates a new api token.
+#### Access
+api-token edit
```
Comment (adrian-enspired) [inline api-token/add.md:9]: access is "logged-in users" (anyone can create an api token for themselves, there is no special permission required).

File: api-token/add.md [hunk 4]
```diff
@@ -1,66 +1,51 @@
...
 
-The new api token is returned in the response payload. This is the only time it will be accessible to the user; it will never be displayed again.
+#### Input
+- string `name` (required): name; must contain a single line of text
```
Comment (adrian-enspired) [inline api-token/add.md:12]: let's say "name for the new token"

File: api-token/add.md [hunk 5]
```diff
@@ -1,66 +1,51 @@
...
+#### Request
 ```
-curl -i -X POST "$PORTAL_API_URL/v1/api-token" \
+$ curl -i -X POST "$PORTAL_API_URL/v1/api-token" \
   -H "Authorization: Bearer $PORTAL_API_KEY" \
   -H "Content-type: application/json" \
   -H "Accept: application/json" \
-  -d '{ "name": "chuck" }'
+  -d '{
+    "name":"Example, Inc."
```
Comment (adrian-enspired) [inline api-token/add.md:21]: the token name is for the user to be able to tell multiple api tokens apart, so let's choose a name that makes sense in that regard. maybe something like "laptop".

File: api-token/add.md [hunk 6]
```diff
@@ -1,66 +1,51 @@
...
 ```
 HTTP/1.1 201 Created
-Server: nginx
-Date: Fri, 12 Jul 2019 17:10:06 GMT
+Date: Tue, 02 Nov 2021 12:51:27 GMT
 Content-Type: application/json;charset=utf-8
-Content-Length: 281
-X-Powered-By: PHP/7.2.15
+Content-Length: 44
+Location: /v1/api-token
```
Comment (adrian-enspired) [inline api-token/add.md:32]: this shouldn't send a `Location` header. did you test and see one?

File: api-token/add.md [hunk 7]
```diff
@@ -1,66 +1,51 @@
...
-Served-By: nwdev-web01-int
 
 {
-  "id": 16,
-  "identity": "chuck",
-  "name": "chuck",
-  "token": "eyJ0eXAiOiJKV1QiL . . . very long auth token"
-}
+  "id": 4,
+  "identity": "Example, Inc.",
```
Comment (adrian-enspired) [inline api-token/add.md:37]: change example name as described above

File: api-token/add.md [hunk 8]
```diff
@@ -1,66 +1,51 @@
...
+  "metadata": {
+    "scope": "client-user-api-token",
+    "uri": "/v1/api-token/4"
+  },
+  "name": "Example, Inc.",
+  "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ2ZXIiOiIyIiwiaWF0IjoxNjM1ODUwMDA3LCJqdGkiOiIzYzAzNjE0NS1iYjE0LTRlZjAtOWIwNy1mN2JkNWU4NDEwYmQiLCJzdWIiOiJ7XCJleHRyYW5ldFwiOntcImNsaWVudFwiOjM4MTE0LFwidXNlclwiOjYxNDIwLFwidG9rZW5faWRcIjo0fX0iLCJhbHYiOjN9._nxUX7XYVX-LJYSuFEdRgpKZnaKcMMzRsIY5o4MazzI"
```
Comment (adrian-enspired) [inline api-token/add.md:43]: truncate this for the example. leave 10 characters or so at the beginning/end and put " ... veryLongAuthToken ..." in the middle.

File: api-token/delete.md [hunk 9]
```diff
@@ -1,56 +1,71 @@
...
+## api-token:delete
+Deletes an Api Token.
 
-api-token:delete
-================
+This task is queued, meaning it will be completed out-of-band from the current request. The response payload will describe the requested task, and will also include a Location header that can be polled to determine the status of the task. @see task:show.
 
-Deletes an api token.
+#### Access
+api-token delete
```
Comment (adrian-enspired) [inline api-token/delete.md:9]: as with create, anyone can delete their own tokens so long as they're logged in. should be "logged-in users".

File: api-token/edit.md [hunk 10]
```diff
@@ -1,56 +1,53 @@
...
+#### Request
 ```
-curl -i -X POST "$PORTAL_API_URL/v1/api-token/$TOKEN_ID" \
+$ curl -i -X POST "$PORTAL_API_URL/v1/api-token" \
   -H "Authorization: Bearer $PORTAL_API_KEY" \
+  -H "Content-type: application/json" \
   -H "Accept: application/json" \
-  -d '{ "name": "charles" }'
+  -d '{
+    "id":"4",
```
Comment (adrian-enspired) [inline api-token/edit.md:22]: the token `id` is not passed in the request body; it must be in the URL.

File: api-token/edit.md [hunk 11]
```diff
@@ -1,56 +1,53 @@
...
+    "name":"Example, Inc"
```
Comment (adrian-enspired) [inline api-token/edit.md:23]: let's use a more descriptive example name, e.g., "workstation".

File: api-token/edit.md [hunk 12]
```diff
@@ -1,56 +1,53 @@
...
-**Access**: logged-in users
-
-**Parameters**:
-- `name`: Required: new name for the token.
-
-**Request**:
+#### Request
 ```
-curl -i -X POST "$PORTAL_API_URL/v1/api-token/$TOKEN_ID" \
+$ curl -i -X POST "$PORTAL_API_URL/v1/api-token" \
```
Comment (adrian-enspired) [inline api-token/edit.md:17]: URL should be `"$PORTAL_API_URL/v1/api-token/$TOKEN_ID"`

File: api-token/edit.md [hunk 13]
```diff
@@ -1,56 +1,53 @@
...
-  -d '{ "name": "charles" }'
+  -d '{
+    "id":"4",
+    "name":"Example, Inc"
+  }'
 ```
 
-**Success Response**: 202 Accepted
+#### Responses
+**Success Response** (request was created for processing): 201 Created
```
Comment (adrian-enspired) [inline api-token/edit.md:28]: should be "request was accepted for processing"
and "202 Accepted"

File: api-token/edit.md [hunk 14]
```diff
@@ -1,56 +1,53 @@
...
-  "identity": "api-token:edit (pending)",
-  "uuid": "34595a2b-40b8-4021-9f0d-6f3e96c8462a",
-  "action": "api-token:edit",
-  "status": "pending",
-  "request_date": 1562959696,
-  "resource_uri": "",
-  "staff_user": null,
-  "user": { "id": 27767, "identity": "Annabel Whyman - user@nocworx.com" }
-}
+  "id": 5,
```
Comment (adrian-enspired) [inline api-token/edit.md:38]: this is the wrong response payload. restore the original "api task" payload.

File: api-token/list.md [hunk 15]
```diff
@@ -1,69 +1,58 @@
...
-- `filter[name]`: Optional: filters results by token name.
-- `filter[id]`: Optional: filters results by token id.
-- `page`: Optional: 1-based result set count for paginated results.
-- `pageSize`: Optional: maximum number of results to include per "page" of a paginated list.
-- `sortBy`: Optional: field to sort the list by; one of `id`|`name`.
-- `sortOrder`: Optional: sort direction; one of `ASC`|`DESC`.
-
-**Request**:
+#### Access
+api-token view
```
Comment (adrian-enspired) [inline api-token/list.md:7]: should be "logged-in users"

File: api-token/edit.md [hunk 16]
```diff
@@ -1,56 +1,53 @@
...
+## api-token:edit
+Edits a stored Api Token.
 
-api-token:show
-==============
+This task is queued, meaning it will be completed out-of-band from the current request. The response payload will describe the requested task, and will also include a Location header that can be polled to determine the status of the task. @see task:show.
 
-Edits an api token.
+#### Access
+api-token edit
```
Comment (adrian-enspired) [inline api-token/edit.md:9]: should be "logged-in users"

File: api-token/list.md [hunk 17]
```diff
@@ -1,69 +1,58 @@
...
+
+#### Input
+- integer `filter[id]` (optional): filter list by system ID.
+- string `filter[name]` (optional): filter list by name.
+- integer `match[id]` (optional): match against id value.
```
Comment (adrian-enspired) [inline api-token/list.md:12]: > match against system ID

File: api-token/list.md [hunk 18]
```diff
@@ -1,69 +1,58 @@
...
+- string `match[name]` (optional): match against name value.
```
Comment (adrian-enspired) [inline api-token/list.md:13]: > match against token name

File: api-token/list.md [hunk 19]
```diff
@@ -1,69 +1,58 @@
...
+- integer `pageSize` (optional): maximum number of results to include per "page" of a paginated list.
+- string `sortBy` (optional): field to sort the list by; one of `id`|`name`.
+- string `sortOrder` (optional): sort direction; one of `ASC`|`DESC`.
+
+#### Request
 ```
-curl -i "$PORTAL_API_URL/v1/api-token" \
+$ curl -i "$PORTAL_API_URL/v1/api-token" \
   -H "Authorization: Bearer $PORTAL_API_KEY" \
+  -H "Content-type: application/json" \
```
Comment (adrian-enspired) [inline api-token/list.md:24]: this example doesn't have a request body; `Content-type` is unneeded

File: api-token/show.md [hunk 20]
```diff
@@ -1,44 +1,43 @@
...
+# Portal: Api Token
 
-**since** 0.0.0
+## api-token:show
+Shows Api Token details.
 
-api-token:show
-==============
+#### Access
+api-token view
```
Comment (adrian-enspired) [inline api-token/show.md:7]: should be "logged-in users"

File: api-token/show.md [hunk 21]
```diff
@@ -1,44 +1,43 @@
...
-
-**Access**: logged-in users
-
-**Parameters**: none
-
-**Request**:
+#### Request
 ```
-curl -i "$PORTAL_API_URL/v1/api-token/$TOKEN_ID" \
+$ curl -i -X GET "$PORTAL_API_URL/v1/api-token/{id}" \
```
Comment (adrian-enspired) [inline api-token/show.md:11]: don't `-X GET` (GET is the default)

URL should be `"$PORTAL_API_URL/v1/api-token/$TOKEN_ID"`. using `$VARIABLES` allows users to easily try things out by defining those vars in their terminal, and then copy+pasting the example.

File: api-token/show.md [hunk 22]
```diff
@@ -1,44 +1,43 @@
...
 ```
 HTTP/1.1 200 OK
-Server: nginx
-Date: Fri, 12 Jul 2019 19:24:32 GMT
+Date: Tue, 02 Nov 2021 12:51:27 GMT
 Content-Type: application/json;charset=utf-8
-Content-Length: 49
-X-Powered-By: PHP/7.2.15
+Content-Length: 44
+Location: /v1/api-token/2
```
Comment (adrian-enspired) [inline api-token/show.md:23]: "show" endpoints do not return `Location` headers

File: api-token/show.md [hunk 23]
```diff
@@ -1,44 +1,43 @@
...
 NocWorx-Api-Version: 0.0.0
-Served-By: nwdev-web01-int
 
 {
-  "id": 14,
-  "identity": "chuck",
-  "name": "chuck"
-}
+  "id": 2,
+    "identity": "Example, Inc.",
```
Comment (adrian-enspired) [inline api-token/show.md:28]: use a token name that matches examples above (e.g., "laptop").

**Review by adrian-enspired — CHANGES_REQUESTED** [review]

File: api-token/add.md (hunk 1 above)
Comment (Neeraja82) [reply api-token/add.md]: Removed, please review. Actually referring to other PR's, which I worked I made these changes for example Address documentation PR.

File: api-token/add.md (hunk 2 above)
Comment (Neeraja82) [reply api-token/add.md:11]: Added back, please review.

File: api-token/add.md (hunk 3 above)
Comment (Neeraja82) [reply api-token/add.md]: Changed, please review. Actually referring to other PR's, which I worked I made these changes for example Address documentation PR.

File: api-token/add.md (hunk 5 above)
Comment (Neeraja82) [reply api-token/add.md]: Changed, please review.

File: api-token/add.md (hunk 6 above)
Comment (Neeraja82) [reply api-token/add.md]: Removed, please review. Actually referring to other PR's, which I worked I made these changes for example Address documentation PR.

File: api-token/add.md (hunk 7 above)
Comment (Neeraja82) [reply api-token/add.md:36]: Changed, please review.

File: api-token/add.md (hunk 8 above)
Comment (Neeraja82) [reply api-token/add.md]: Changed, please review.

File: api-token/delete.md (hunk 9 above)
Comment (Neeraja82) [reply api-token/delete.md]: Changed, please review.

File: api-token/edit.md (hunk 10 above)
Comment (Neeraja82) [reply api-token/edit.md]: Yes changed, please review.

File: api-token/edit.md (hunk 11 above)
Comment (Neeraja82) [reply api-token/edit.md]: Changed, please review.

File: api-token/edit.md (hunk 12 above)
Comment (Neeraja82) [reply api-token/edit.md]: Yes, changed. Please review.

File: api-token/edit.md (hunk 13 above)
Comment (Neeraja82) [reply api-token/edit.md]: Changed, please review.

File: api-token/edit.md (hunk 14 above)
Comment (Neeraja82) [reply api-token/edit.md]: Added task api  response payload. Please review.

File: api-token/list.md (hunk 15 above)
Comment (Neeraja82) [reply api-token/list.md]: Changed, please review.

File: api-token/edit.md (hunk 16 above)
Comment (Neeraja82) [reply api-token/edit.md]: Changed, please review.

File: api-token/list.md (hunk 17 above)
Comment (Neeraja82) [reply api-token/list.md]: Changed, please review.

File: api-token/list.md (hunk 18 above)
Comment (Neeraja82) [reply api-token/list.md]: Changed, please review.

File: api-token/list.md (hunk 19 above)
Comment (Neeraja82) [reply api-token/list.md]: Changed, please review.

File: api-token/show.md (hunk 21 above)
Comment (Neeraja82) [reply api-token/show.md]: Changed, please review.

File: api-token/show.md (hunk 22 above)
Comment (Neeraja82) [reply api-token/show.md]: Changed, please review. Actually referring to other PR's, which I worked I made these changes for example Address documentation PR.

File: api-token/show.md (hunk 23 above)
Comment (Neeraja82) [reply api-token/show.md]: Changed, please review.

**Comment by Neeraja82** [comment]
Addressed all the review suggestions. Please review.

**Review by adrian-enspired — APPROVED** [review]

**Pull request merged**